lib/os/ring_buffer.c

```diff
@@ -197,11 +205,21 @@ uint32_t ring_buf_put_claim(struct ring_buf *buf, uint8_t **data, uint32_t size)
 
 int ring_buf_put_finish(struct ring_buf *buf, uint32_t size)
 {
+	uint32_t rew;
+	uint32_t threshold = ring_buf_get_rewind_threshold();
+
 	if ((buf->tail + size) > (buf->head + buf->size)) {
```

Comment:
Will this calculation also need to take into account that `buf->head` may have been rewound?

Reply:
No, because tail is always first one to be rewound. Head just follows based on detection that it is bigger than tail.